lib/services/local/saml_idp_service_provider.go

```diff
@@ -183,3 +208,67 @@ func validateSAMLIdPServiceProvider(sp types.SAMLIdPServiceProvider) error {
 
 	return nil
 }
+
+// fetchAndSetEntityDescriptor fetches Service Provider entity descriptor (aka SP metadata)
+// from remote metadata endpoint (Entity ID) and sets it to sp if the xml format
+// is a valid Service Provider metadata format.
+func (s *SAMLIdPServiceProviderService) fetchAndSetEntityDescriptor(sp types.SAMLIdPServiceProvider) error {
+	var httpClient *http.Client
+	if s.httpClient != nil {
+		httpClient = s.httpClient
+	} else {
+		httpClient = &http.Client{
+			Timeout: defaults.HTTPRequestTimeout,
+			CheckRedirect: func(req *http.Request, via []*http.Request) error {
+				return http.ErrUseLastResponse
+			},
+		}
+	}
```

Comment:
It feels like this bit should go in `NewSAMLIdPServiceProviderService` so that we don't have to create a new one each time given that nothing actually supplies this at present.

Reply:
Oh yes, thanks for pointing out. I started here, then added `*http.Client` to `NewSAMLIdPServiceProviderService` but forgot to move this up there.

Reply:
Moved to `NewSAMLIdPServiceProviderService ` in https://github.com/gravitational/teleport/pull/35454/commits/2a20721c4b4a535b095a2837ca60b9d05864fbdd